lib/project_types/script/layers/infrastructure/languages/other_project_creator.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+module Script
+  module Layers
+    module Infrastructure
+      module Languages
+        class OtherProjectCreator < ProjectCreator
+          def self.config_file
+            "script.config.yml"
+          end
+
+          def self.metadata_file
+            "metadata.json"
+          end
+
+          def setup_dependencies
+            generate_config
+            generate_metadata
+          end
+
+          def create_start_message
+            ctx.message(
+              "script.create.preparing_project",
+              sparse_checkout_repo,
+              project_name
+            )
+          end
+
+          def create_inprogress_message
+            ctx.message(
+              "script.create.creating_other",
+              sparse_checkout_repo,
+              project_name
+            )
+          end
+
+          def create_finished_message
+            ctx.message("script.create.created_other", project_name)
+          end
+
+          private
+
+          # the config is equivalent to TS's script.config.yml
+          # ex: https://github.com/Shopify/scripts-apis-examples/blob/master/checkout/typescript/payment-methods/default/script.config.yml
+          def generate_config
```

Comment:
Nit: I believe instead of storing the string representations of the config and metadata, you could keep it as a hash in the code and then call the `to_json` or `to_yaml`. The cost of doing this isn't very high and makes the code more readable IMO.